invoke/tasks.py

```diff
@@ -116,6 +116,38 @@ def __call__(self, *args, **kwargs):
     def called(self):
         return self.times_called > 0
 
+    def get_arg_names_and_spec_dict(self, spec):
```

Comment:
I normally would define helper functions with a `_` prefix _after_ the definition of the caller. In this case, I probably would make it a free function, instead of a function on the class. 

However, I followed the precedent of having helper methods come before the caller methods, and having no `_` prefix. Please let me know if you'd prefer the other style.